File: 2624-difference-between-element-sum-and-digit-sum-of-an-array/2624-difference-between-element-sum-and-digit-sum-of-an-array.cpp
class Solution {
public:
    int differenceOfSum(vector<int>& nums) {
        int elem_sum = 0, digit_sum = 0;
        for (auto& num : nums) {
            elem_sum += num;
            for (auto& ch : to_string(num)) {
                digit_sum += ch - '0';
            }
        }
        return elem_sum - digit_sum;
    }
};

File: 2624-difference-between-element-sum-and-digit-sum-of-an-array/2624-difference-between-element-sum-and-digit-sum-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var differenceOfSum = function(nums) {
    const sum = (arr) => arr.reduce((acc, curr) => acc + curr)
    const [elemSum, digitSum] = [nums, nums.map(addDigits)].map(sum)
    return elemSum - digitSum
};

function addDigits(num) {
    let sum = 0
    while (num > 0) {
        sum += num % 10
        num = Math.floor(num / 10)
    }
    return sum
}